martha/martha_v3.js

```diff
@@ -478,87 +489,130 @@ async function retrieveFromServers(params) {
-            const accessTokenResponse = await apiAdapter.getJsonFrom(bondAccessTokenUrl, auth);
-            accessToken = accessTokenResponse.token;
+            if (overlapFields(requestedFields, MARTHA_V3_ACCESS_ID_FIELDS)) {
+                try {
+                    accessId = getDrsAccessId(drsResponse, accessMethodType);
+                } catch (error) {
+                    throw new RemoteServerError(error, 'Received error while parsing the access id.');
+                }
+            }
+
+            // Retrieve an accessToken from Bond that will be used to later retrieve the accessUrl from the DRS server.
+            let accessToken;
+            if (bondProvider && accessId) {
+                try {
+                    const bondAccessTokenUrl = `${config.bondBaseUrl}/api/link/v1/${bondProvider}/accesstoken`;
+                    console.log(`Requesting Bond access token for '${url}' from '${bondAccessTokenUrl}'`);
+                    const accessTokenResponse = await apiAdapter.getJsonFrom(bondAccessTokenUrl, auth);
+                    accessToken = accessTokenResponse.token;
+                } catch (error) {
+                    throw new RemoteServerError(error, 'Received error contacting Bond.');
+                }
+            }
+
+            // Retrieve the accessUrl using the returned accessToken, even if the token was empty.
+            if (bondProvider && accessId) {
+                try {
+                    const httpsAccessUrl = generateAccessUrl(drsType, accessId);
+                    const accessTokenAuth = `Bearer ${accessToken}`;
+                    console.log(`Requesting DRS access URL for '${url}' from '${httpsAccessUrl}'`);
+                    accessUrl = await apiAdapter.getJsonFrom(httpsAccessUrl, accessTokenAuth);
+                } catch (error) {
+                    throw new RemoteServerError(error, 'Received error contacting DRS provider.');
+                }
+            }
         } catch (error) {
-            throw new RemoteServerError(error, 'Received error contacting Bond.');
+            // Just logging the error for now. Eventually, we'll want to remove this outer try.
+            console.warn('Ignoring error from fetching signed URL:', error);
         }
-    }
+    };
 
-    // Retrieve the accessUrl using the returned accessToken, even if the token was empty.
-    let accessUrl;
-    if (bondProvider && accessId) {
-        try {
-            const httpsAccessUrl = generateAccessUrl(drsType, accessId);
-            const accessTokenAuth = `Bearer ${accessToken}`;
-            console.log(`Requesting DRS access URL for '${url}' from '${httpsAccessUrl}'`);
-            accessUrl = await apiAdapter.getJsonFrom(httpsAccessUrl, accessTokenAuth);
-        } catch (error) {
-            throw new RemoteServerError(error, 'Received error contacting DRS provider.');
-        }
+    let ranOutOfTime;
```

Comment:
I was getting used to `let`s last night and forgot that I could instead return something from this function (and `fetch`) to indicate which promise won the race. I'll probably add a commit to this PR to clean this up.